huaweicloud/config/hc_config.go

```diff
@@ -134,92 +135,49 @@ func getServiceEndpoint(c *Config, srv, region string) string {
 }
 
 // NewVpcClient is the VPC service client using huaweicloud-sdk-go-v3 package
-func NewVpcClient(c *Config, region string) (*vpc.VpcClient, error) {
-	credentials, err := buildAuthCredentials(c, region)
+func (c *Config) HcVpcV3Client(region string) (*vpcv3.VpcClient, error) {
+	hcClient, err := NewHcClient(c, region, "vpc", false)
 	if err != nil {
 		return nil, err
 	}
 
-	vpcEndpoint := getServiceEndpoint(c, "vpc", region)
-	if vpcEndpoint == "" {
-		return nil, fmt.Errorf("failed to get the endpoint of VPC service")
-	}
-
-	return vpc.NewVpcClient(
-		vpc.VpcClientBuilder().
-			WithEndpoint(vpcEndpoint).
-			WithCredential(*credentials).
-			WithHttpConfig(buildHTTPConfig(c)).
-			Build()), nil
+	return vpcv3.NewVpcClient(hcClient), nil
 }
 
 // NewTmsClient is the TMS service client using huaweicloud-sdk-go-v3 package
-func NewTmsClient(c *Config, region string) (*tms.TmsClient, error) {
-	credentials, err := buildGlobalAuthCredentials(c, region)
+func (c *Config) HcTmsV1Client(region string) (*tmsv1.TmsClient, error) {
+	hcClient, err := NewHcClient(c, region, "tms", true)
 	if err != nil {
 		return nil, err
 	}
-
-	tmsEndpoint := getServiceEndpoint(c, "tms", region)
-	if tmsEndpoint == "" {
-		return nil, fmt.Errorf("failed to get the endpoint of TMS service")
-	}
-
-	return tms.NewTmsClient(
-		tms.TmsClientBuilder().
-			WithEndpoint(tmsEndpoint).
-			WithCredential(*credentials).
-			WithHttpConfig(buildHTTPConfig(c)).
-			Build()), nil
+	return tmsv1.NewTmsClient(hcClient), nil
 }
 
 // NewKmsClient is the KMS service client using huaweicloud-sdk-go-v3 package
-func NewKmsClient(c *Config, region string) (*kps.KpsClient, error) {
-	credentials, err := buildAuthCredentials(c, region)
+func (c *Config) HcKmsV3Client(region string) (*kpsv3.KpsClient, error) {
+	hcClient, err := NewHcClient(c, region, "kmsv3", false)
```

Comment:
if we just use the endpoing, do we need kmsv3?

Reply:
ok，deleted